src/main/java/com/systemmeltdown/robotlib/subsystems/SingleSpeedTalonDriveSub.java

```diff
@@ -0,0 +1,194 @@
+package com.systemmeltdown.robotlib.subsystems;
+
+import java.util.ArrayList;
+
+import com.ctre.phoenix.motorcontrol.ControlMode;
+import com.ctre.phoenix.motorcontrol.FeedbackDevice;
+import com.ctre.phoenix.motorcontrol.NeutralMode;
+import com.ctre.phoenix.motorcontrol.StatusFrame;
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonSRX;
+import com.systemmeltdown.robotlib.util.PIDValues;
+import com.systemmeltdown.robotlib.util.Utility;
+
+public class SingleSpeedTalonDriveSub extends SkidSteerDriveSubBase {
+    WPI_TalonSRX m_rightMaster;
+    WPI_TalonSRX m_leftMaster;
+    WPI_TalonSRX[] m_rightSlaves;
+    WPI_TalonSRX[] m_leftSlaves;
+    int m_timeout;
+    int m_talonPidPrimary;
+    double m_driveMotorDeadband;
+    double m_driveRampSeconds;
+
+    int m_talonSlotVelocity;
+    int m_talonSlotDistance;
+    int m_talonSlotTurning;
+    PIDValues m_pidDrivePos;
+
+    PIDValues m_pidDriveSpeed;
+
+    public SingleSpeedTalonDriveSub(int rightMasterID, int leftMasterID, int[] rightSlaveIDS, int[] leftSlaveIDS,
+            int timeout, int talonPidPrimary, double driveMotorDeadband, double driveRampSeconds, int talonSlotVelocity,
+            int talonSlotDistance, int talonSlotTurning, PIDValues pidDrivePos, PIDValues pidDriveSpeed) {
+
+        ArrayList<WPI_TalonSRX> rightSlaves = new ArrayList<>();
+        ArrayList<WPI_TalonSRX> leftSlaves = new ArrayList<>();
+
+        for (int id : rightSlaveIDS) {
+            rightSlaves.add(new WPI_TalonSRX(id));
+        }
+
+        for (int id : leftSlaveIDS) {
+            leftSlaves.add(new WPI_TalonSRX(id));
+        }
+
+        m_rightMaster = new WPI_TalonSRX(rightMasterID);
+        m_leftMaster = new WPI_TalonSRX(leftMasterID);
+        m_rightSlaves = rightSlaves.toArray(m_rightSlaves);
+        m_leftSlaves = leftSlaves.toArray(m_leftSlaves);
+
+        m_timeout = timeout;
+        m_talonPidPrimary = talonPidPrimary;
+        m_driveMotorDeadband = driveMotorDeadband;
+        m_driveRampSeconds = driveRampSeconds;
+
+        m_talonSlotVelocity = talonSlotVelocity;
+        m_talonSlotDistance = talonSlotDistance;
+        m_talonSlotTurning = talonSlotTurning;
+        m_pidDrivePos = pidDrivePos;
+
+        m_pidDriveSpeed = pidDriveSpeed;
+
+    }
+
+    public void configure() {
```

Comment:
It would be great if we took a configuration object in the constructor above and then passed that configuration object here.